### ISSUE TITLE
pSet Properties is not working.

### ISSUE THREAD

**Issue by shailendra26-siso** [IFCjs/components#101]
### Describe the bug 📝

https://github.com/IFCjs/components/blob/big-restructure/src/ifc/IfcPropertiesUtils/index.ts#L8
I am working with this feature but not get the success. Can you help me on this please?
I am using the **Advance IFC JS course for implementation. **

There is error when it's matching the type of property. Somewhere "IFCPROJECT" is matching with type.
Used the Defined the functions for pSet:

![image](https://github.com/IFCjs/components/assets/119593005/fd86dae3-d872-4a40-a186-5ed244276e07)

![image](https://github.com/IFCjs/components/assets/119593005/59a427a4-d42a-457c-bc75-fc8be025c1fb)

I tried with static expressID but it is also giving the error, see above screenshots.

![image](https://github.com/IFCjs/components/assets/119593005/54778b1d-6867-4229-8009-fcaa95473fad)

![image](https://github.com/IFCjs/components/assets/119593005/c8ef47d3-997d-44b5-950e-65e8e3027af0)

![image](https://github.com/IFCjs/components/assets/119593005/8d159eb3-e550-439d-8780-75cb0bda625e)

![image](https://github.com/IFCjs/components/assets/119593005/b1339623-3a2f-41fd-9f95-90f55e688e91)

### System Info 💻

```shell
openbim-components
```

### Used Package Manager 📦

npm

### Validations ✅

- [x] Read the [docs](https://ifcjs.github.io/info/docs/Introduction).
- [x] Check that there isn't [already an issue](https://github.com/IFCjs/components/issues) that reports the same bug to avoid creating a duplicate.
- [x] Make sure this is a IFC.js components issue and not a framework-specific issue. For example, if it's a THREE.js related bug, it should likely be reported to [mrdoob/threejs](https://github.com/mrdoob/three.js) instead.
- [x] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/FXfyR4XrKT).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by agviegas** [comment]
Is this a duplicate of #100?

**Comment by shailendra26-siso** [comment]
Hi,

 

pSet Properties is not working. · Issue #101 · IFCjs/components · GitHub <https://github.com/IFCjs/components/issues/101>  and pSet Properties is not working. · Issue #100 · IFCjs/components · GitHub <https://github.com/IFCjs/components/issues/100>  is related. Yes you can say that It is same.

 

Shailendra Singh

 



SISO Software’s India Pvt. Ltd.

B-3/61, Safdarjung Enclave, IN-110029 Delhi, India.

Contact: +91-9911682739

 <http://www.siso.net/> www.siso.net

**Comment by agviegas** [comment]
Ok, got it. I already mentioned this in another issue, but can you please answer in Github directly instead of answering your email? Your messages are getting bigger and bigger and it might make this a bit hard to follow. We would also need a full minimal application that we can run where we can reproduce this to fix it. Thanks a lot in advance!

**Comment by HoyosJuan** [comment]
@shailendra26-siso it's a bit hard to tell what is going on in your code, but a couple of things to keep in mind:

1. In other cases you've said you're still using web-ifc-viewer. Keep in mind, web-ifc-viewer and openbim-components are not compatible between them, in other words you can't work with them interchangeably in the same codebase.
2. In openbim-components, the IFC entity types are not strings, but numbers. So, if you want to take all entities from a particular type, then you would have to find it by its type numbers (you can find that in web-ifc).
3. Once again, it's a bit hard to know what is going on in your code, but from the screenshots you have shown the error tell me you are working with a variable named `t` that you are not defining anywhere.

Hope this helps somehow.

**Comment by HoyosJuan** [comment]
This is solved in PR https://github.com/IFCjs/components/pull/214.
